### PR TITLE
Merge darwin/trunk, including libdispatch-703.1.4 Apple source drop, to swift-3.0-branch

### PR DESCRIPTION
**Pull request by das**
Merge tag darwin/darwin-006 on darwin/trunk, which includes the latest Apple source drop libdispatch-703.1.4 as well as with all the github PR’s that we’ve pulled back into our internal repository since the last darwin/trunk merge, along with any changes made to these during internal review and testing on Darwin.

### CODE REVIEW

**Comment by das** [comment]
@swift-ci Please test

**Comment by das** [comment]
@swift-ci Please test

**Comment by das** [comment]
@RLovelett that seems like a complaint about https://github.com/apple/swift-corelibs-libdispatch/pull/139 and not this PR ? that flag was introduced there

**Comment by RLovelett** [comment]
**tl;dr** I did not intend to bring the nitpick up in this PR. In fact, I didn't know it even existed when I wrote my comment. 😄 

@das you are right. I wrote the comment on the line of code in the change for #139 (specifically https://github.com/apple/swift-corelibs-libdispatch/commit/b65aa22fbd84907a9f3d570e695cb0b3e482a1a3).

GitHub pulls comments on lines of code in the comments section for a PR. Because you are pulling to a new branch that contains a commit (https://github.com/apple/swift-corelibs-libdispatch/commit/b65aa22fbd84907a9f3d570e695cb0b3e482a1a3) that I commented on GitHub brought the comment into this PR thread.

I certainly do not intend to have my nitpick slow-down the swift-3.0-branch.

Sorry for the trouble. 🖖

**Comment by das** [comment]
@RLovelett: no problem, thanks

**Comment by das** [comment]
@swift-ci Please test
